Provide more precise description for Lockdown Mode Alert in MacCatalyst

brentfulgham opened the pull request:
#### 7a7a1dfaa36dd3ab2e38e3982f3766ff6055ee8a
<pre>
Provide more precise description for Lockdown Mode Alert in MacCatalyst
<a href="https://bugs.webkit.org/show_bug.cgi?id=256005">https://bugs.webkit.org/show_bug.cgi?id=256005</a>
&lt;rdar://101691889&gt;

Reviewed by Aditya Keerthi.

MacCatalyst apps present the same Lockdown Mode Alert as the one they use in iOS,
which is misleading. This change revises the text we use in MacCatalyst to be more
precise about the features available for controlling Lockdown Mode.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _presentLockdownMode]):

Canonical link: <a href="https://commits.webkit.org/263470@main">https://commits.webkit.org/263470@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5a1ed3b4fd80d73c97c42444dd7700d3d1f0a71d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4704 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4824 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4984 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6208 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4856 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4698 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4974 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4790 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5083 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4781 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4869 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4212 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6218 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2358 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4203 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9185 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4217 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4277 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5845 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4685 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3817 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4209 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1154 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8257 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4569 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
```diff
@@ -3125,7 +3125,11 @@ - (void)_presentLockdownMode
     if (!lockdownModeWarningNeeded)
         return;
 
+#if PLATFORM(IOS_FAMILY) && !PLATFORM(MACCATALYST)
```
No need for `PLATFORM(IOS_FAMILY)` since the entire file is guarded by that.

Would re-write as:

```
#if PLATFORM(MACCATALYST)
#else
#endif
```

brentfulgham replied:
Will do!

brentfulgham committed:
Provide more precise description for Lockdown Mode Alert in MacCatalyst
https://bugs.webkit.org/show_bug.cgi?id=256005
<rdar://101691889>

Reviewed by Aditya Keerthi.

MacCatalyst apps present the same Lockdown Mode Alert as the one they use in iOS,
which is misleading. This change revises the text we use in MacCatalyst to be more
precise about the features available for controlling Lockdown Mode.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _presentLockdownMode]):

Canonical link: https://commits.webkit.org/263470@main
